blc/checker.py

```diff
@@ -226,8 +232,20 @@ def update_list(self, response: requests.Response) -> None:
     def run(self) -> None:
         """Run the checker."""
         # We check while we have an URL unchecked
-        while (self.url_to_check):
-            response = self.check(self.url_to_check.pop(0))
-            if response:
-                self.update_list(response)
-            time.sleep(self.delay)
+        while 1:
+            url_to_check = [u for u in self.urls if not self.urls[u]['result']]
```

Comment:
```suggestion
            url_to_check = [u for u in self.urls if not self.urls[u].get('result', False)]
```
to prevent a KeyError raising here !